src/index.js

```diff
@@ -86,116 +108,146 @@ export default function nodeResolve ( options = {} ) {
 			// disregard entry module
 			if ( !importer ) return null;
 
-			if (options.browser && browserMapCache[importer]) {
-				const resolvedImportee = resolve( dirname( importer ), importee );
-				const browser = browserMapCache[importer];
-				if (browser[importee] === false || browser[resolvedImportee] === false) {
-					return ES6_BROWSER_EMPTY;
-				}
-				if (browser[importee] || browser[resolvedImportee] || browser[resolvedImportee + '.js'] || browser[resolvedImportee + '.json']) {
-					importee = browser[importee] || browser[resolvedImportee] || browser[resolvedImportee + '.js'] || browser[resolvedImportee + '.json'];
-				}
-			}
-
-
-			const parts = importee.split( /[/\\]/ );
-			let id = parts.shift();
-
-			if ( id[0] === '@' && parts.length ) {
-				// scoped packages
-				id += `/${parts.shift()}`;
-			} else if ( id[0] === '.' ) {
-				// an import relative to the parent dir of the importer
-				id = resolve( importer, '..', importee );
-			}
-
-			if (only && !only.some(pattern => pattern.test(id))) return null;
-
-			let disregardResult = false;
-			let packageBrowserField = false;
-			const extensions = options.extensions || DEFAULT_EXTS;
-
-			const resolveOptions = {
-				basedir: dirname( importer ),
-				packageFilter ( pkg, pkgPath ) {
-					const pkgRoot = dirname( pkgPath );
-					if (options.browser && typeof pkg[ 'browser' ] === 'object') {
-						packageBrowserField = Object.keys(pkg[ 'browser' ]).reduce((browser, key) => {
-							const resolved = pkg[ 'browser' ][ key ] === false ? false : resolve( pkgRoot, pkg[ 'browser' ][ key ] );
-							browser[ key ] = resolved;
-							if ( key[0] === '.' ) {
-								const absoluteKey = resolve( pkgRoot, key );
-								browser[ absoluteKey ] = resolved;
-								if ( !extname(key) ) {
-									extensions.reduce( ( browser, ext ) => {
-										browser[ absoluteKey + ext ] = browser[ key ];
-										return browser;
-									}, browser );
-								}
-							}
-							return browser;
-						}, {});
+			return Promise.resolve(this.cache ? this.cache.get(`${importee}|${importer}`) : false)
```

Comment:
I wonder if we should have cache loading async, or if we can just assume that the initial async cache load has already completed so that the cache is always synchronous.

The only time this might be an issue is if the cache grows very large, and we want to load the cache itself partially with an index or something. But on a per-project level I struggle to see us getting there? I mean loading a few MB into memory shouldn't be an issue right?

Reply:
I had considered this - but even for some really overblown cache we're still going to be looking at sub 100mb cache sizes. This plugin will to store around ~100bytes per entry on average for maybe 1000-2000 entries for a larger build. So just this plugin will maybe store 10-20kb of cache. The transform cache could be larger - perhaps in the megabytes as it has to store input code and output code; but cache in the megabytes will still be quick enough to load from disk before any plugins are called.

Reply:
Ok agreed. In that case, I think we can assume `this.cache` is available synchronously for plugins.